src/pages/history/history.tsx

```diff
@@ -1,13 +1,191 @@
+'use client';
+
+import Box from '@mui/material/Box';
+import Button from '@mui/material/Button';
+import Card from '@mui/material/Card';
+import CardActionArea from '@mui/material/CardActionArea';
 import Container from '@mui/material/Container';
+import Stack from '@mui/material/Stack';
 import Typography from '@mui/material/Typography';
+import { useTranslations } from 'next-intl';
+import { useEffect, useState } from 'react';
+
+import Loader from '@/components/layout/loader/loader';
+import { useAppSelector } from '@/hooks/redux';
+import { Link } from '@/i18n/navigation';
+import type { HistoryCollection } from '@/types/userData';
+import { getHistory } from '@/utils/firebase/collections';
+import { sortHistoryCollection } from '@/utils/firebase/sortHistoryCollection';
 
 export default function History() {
+  const t = useTranslations('history_general');
+  const { user } = useAppSelector((state) => state.user);
+  const [requests, setRequests] = useState<HistoryCollection[] | null>(null);
+  const [isLoadingData, setIsLoadingData] = useState(true);
+
+  useEffect(() => {
+    if (!user) return;
+
+    const loadHistory = async () => {
+      setIsLoadingData(true);
+      try {
+        const data = await getHistory(user.userId);
+        setRequests(sortHistoryCollection(data));
+      } catch {
+        setRequests([]);
+      } finally {
+        setIsLoadingData(false);
+      }
+    };
+
+    void loadHistory();
+  }, [user]);
+
+  if (isLoadingData || requests === null) return <Loader />;
```

Comment:
Would that be better?
```suggestion
  if (isLoadingData || !requests ) return <Loader />;
```

Reply:
Yes, I'll fix it in next PR